spec/controllers/admin/manager_invitations_controller_spec.rb

```diff
@@ -0,0 +1,41 @@
+require 'spec_helper'
+
+module Admin
+  describe ManagerInvitationsController, type: :controller do
+    let!(:existing_user) { create(:user) }
+    let!(:enterprise) { create(:enterprise) }
+    let(:admin) { create(:admin_user) }
+
+    describe "#create" do
+      context "when given email matches an existing user" do
+        before do
+          controller.stub spree_current_user: admin
+        end
+
+        it "returns an error" do
+          spree_post :create, {email: existing_user.email, enterprise_id: enterprise.id}
+
+          expect(response.status).to eq 422
+          expect(json_response['errors']).to eq I18n.t('admin.enterprises.invite_manager.user_already_exists')
+        end
+      end
+
+      context "signing up a new user" do
+        before do
+          controller.stub spree_current_user: admin
+        end
+
+        it "creates a new user, sends an invitation email, and returns the user id" do
+          spree_post :create, {email: 'un.registered@email.com', enterprise_id: enterprise.id}
+
+          new_user = Spree::User.find_by_email('un.registered@email.com')
+
+          expect(new_user.reset_password_token).to_not be_nil
+          expect(Delayed::Job.last.payload_object.class.to_s).to eq('ManagerInvitationJob')
```

Comment:
this seems to coupled with the code to me. You don't really care what type it is as long as it does what it says and besides, it may prevent us to change the name of the class in the future.